### DISCUSSION TITLE
Combination of Symmetry and No-slip wall BCs

### DISCUSSION THREAD

**Discussion by SRkumar97** [AMReX-Combustion/PeleC#876] [Q&A] [answered]
Hi,
I want run an external boundary layer case over a flat domain. I want to setup the case in the exact same manner as this schematic below, but for compressible flow. **Image courtesy - NASA LaRC**
![image](https://github.com/user-attachments/assets/6894d6a2-bd3b-4f68-b623-c39a34fc4be6)
Now, at the base of the 3D domain, I will need to use Symmetry for an initial distance, followed by No-slip wall from the region of interest, I wish to know how to combine and set this case up. All I know is that the "Hard" BC type has to be employed here. However, what is the best way to inform the code to smartly use Symmetry and No-Slip wall BCs, in the respective if{}-else{} statements of these regions, in subroutine bcnormal() of prob.H? I believe I'd mess up many things if I chose to manually specify every flow field variable matrix using s_int and s_ext, especially in the viscous no-slip wall region.
Thanks a ton in advance!

**Comment by baperry2** [comment, marked as answer]
This is actually pretty easy to accomplish. `s_ext` that gets passed to bcnormal comes prepopulated with values for a no-slip wall. to specify a symmetry condition (equivalent to a slip wall), all you need to do is change the sign of the tangential components of velocity. See the example in the MassCons case, which shows how to define both slip and no-slip walls in `bcnormal`: https://github.com/AMReX-Combustion/PeleC/blob/4ba05a820a73cb06d81bdd681b313b2ab1a218ac/Exec/RegTests/MassCons/prob.H#L71

Documentation: https://amrex-combustion.github.io/PeleC/BoundaryConditions.html#boundary-conditions

**Reply by SRkumar97** [reply]
Thanks, I'm checking this case. Here, I would just like to know the role of the two python code files - **run_isothermal_study.py** and **run_masscons_study.py**
Are these codes to be run in the post-processing phase? I can't see any mention of these in the README.md file.

**Reply by baperry2** [reply]
Those scripts run a few cases and then process data from them to verify numerical convergence for isothermal walls and mass conservation, respectively. You do not need to worry about them if you don't want to.

**Reply by SRkumar97** [reply]
Ohh, ok. Thanks for the clarification.

**Reply by SRkumar97** [reply]
In the input files of this case, there is this parameter **prob.wall_type** which I have never encountered before, nor do I see any mention of it in the official PeleC documentation site. Does this correspond/replicate the same kind of wall BC as **pelec.do_isothermal_walls**?

**Reply by baperry2** [reply]
Any input prefixed with `prob` is problem specific and defined in that problems case files. In this case, that input parameter is documented for the README for the case, and you can also look in the `prob.{H,cpp}` and `prob_parm.H` files for how it iis defined/used.

**Reply by SRkumar97** [reply]
Thanks, I didn't understand this clearly before.

**Reply by SRkumar97** [reply]
Hi, I have a case in which I need to use a similar combination of BCs, but this time a combination of inlet and outlet. In essence, at the domain top I need to introduce an inflow for some distance, followed by either
(a) a pressure outlet BC using Hard, or 
(b) an extrapolation using FOEXtrap 

for the remaining distance. In case (a), the entire top will essentially have only Hard, but with inlet and pressure outlet whereas in (b) the domain top will have Hard and FOExtrap BCs
I have edited the previous schematic for your reference. How to go about these combinations? 

![image](https://github.com/user-attachments/assets/a70bcb1a-9bae-48c5-9221-b06e2caa45c1)

Thanks a ton in advance!

**Reply by SRkumar97** [reply]
Hi, please correct me if I'm wrong. If I'd like to manually set FOExtrap for some regions within a Hard BC, do I just do the following?

s_ext[URHO] = s_int[URHO];
    s_ext[UMX] = s_int[UMX];
    :
etc. for all the 6 primitive variables?

**Reply by SRkumar97** [reply]
This worked to set FOExtrap BC manually. 

> Hi, please correct me if I'm wrong. If I'd like to manually set FOExtrap for some regions within a Hard BC, do I just do the following?
> 
> s_ext[URHO] = s_int[URHO]; s_ext[UMX] = s_int[UMX]; : etc. for all the 6 primitive variables?

**Reply by baperry2** [reply]
Yes that is the correct strategy for setting FOExtrap boundaries within a user BC, glad it worked for you.
